synapseclient/models/agent.py

```diff
@@ -0,0 +1,629 @@
+            syn.login(silent=True)
+
+            my_agent = Agent(cloud_agent_id=AWS_AGENT_ID)
+            my_agent.register()
+
+            my_agent.prompt(
+                prompt="Hello",
+            enable_trace=True,
+            print_response=True,
+        )
+
+    Advanced Example: Start and prompt multiple sessions
+        Here, we connect to a custom agent and start one session with the prompt "Hello".
+        In the background, this first session is being set as the current session
+        and future prompts will be sent to this session by default. If we want to send a
+        prompt to a different session, we can do so by starting it and calling prompt again,
+        but with our new session as an argument. We now have two sessions, both stored in the
+        `my_agent.sessions` dictionary. After the second prompt, `my_second_session` is now
+        the current session.
+
+            syn = Synapse()
+            syn.login()
+
+            my_agent = Agent(registration_id=my_registration_id).get()
+
+            my_agent.prompt(
+                prompt="Hello",
+                enable_trace=True,
+                print_response=True,
+            )
+
+            my_second_session = my_agent.start_session()
+            my_agent.prompt(
+                prompt="Hello again",
+                enable_trace=True,
+                print_response=True,
+                session=my_second_session,
+            )
+    """
+
+    cloud_agent_id: Optional[str] = None
+    """The unique ID of the agent in the cloud provider."""
+
+    cloud_alias_id: Optional[str] = None
+    """The alias ID of the agent in the cloud provider.
+        Defaults to 'TSTALIASID' in the Synapse API.
+    """
+
+    registration_id: Optional[int] = None
+    """The ID number of the agent assigned by Synapse."""
+
+    registered_on: Optional[datetime] = None
+    """The date the agent was registered."""
+
+    type: Optional[AgentType] = None
+    """The type of agent. One of either BASELINE or CUSTOM."""
+
+    sessions: Dict[str, AgentSession] = field(default_factory=dict)
+    """A dictionary of AgentSession objects, keyed by session ID."""
+
+    current_session: Optional[AgentSession] = None
+    """The current session. Prompts will be sent to this session by default."""
+
+    def fill_from_dict(self, agent_registration: Dict[str, str]) -> "Agent":
+        """
+        Converts a response from the REST API into this dataclass.
+
+        Arguments:
+            agent_registration: The response from the REST API.
+
+        Returns:
+            The Agent object.
+        """
+        self.cloud_agent_id = agent_registration.get("awsAgentId", None)
+        self.cloud_alias_id = agent_registration.get("awsAliasId", None)
+        self.registration_id = agent_registration.get("agentRegistrationId", None)
+        self.registered_on = agent_registration.get("registeredOn", None)
+        self.type = agent_registration.get("type", None)
```

Comment:
This should be updated to create an Enum of `AgentSession`. Right now this would set `self.type` to a type `str`